code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm

```diff
@@ -50,6 +49,56 @@ Difficulty: Medium
 	elimination = 1
 	appearance_flags = 0
 	mouse_opacity = MOUSE_OPACITY_ICON
+	attack_action_types = list(/datum/action/innate/megafauna_attack/create_skull,
+							   /datum/action/innate/megafauna_attack/charge_target)
+	small_sprite_type = /datum/action/small_sprite/megafauna/legion
+
+/datum/action/innate/megafauna_attack/create_skull
+	name = "Create Legion Skull"
+	icon_icon = 'icons/mob/lavaland/lavaland_monsters.dmi'
+	button_icon_state = "legion_head"
+	chosen_message = "<span class='colossus'>You are now creating legion skulls.</span>"
+	chosen_attack_num = 1
+
+/datum/action/innate/megafauna_attack/charge_target
+	name = "Charge Target"
+	icon_icon = 'icons/mob/actions/actions_items.dmi'
+	button_icon_state = "sniper_zoom"
+	chosen_message = "<span class='colossus'>You are now charging at your target.</span>"
+	chosen_attack_num = 2
+
+/mob/living/simple_animal/hostile/megafauna/legion/OpenFire(the_target)
+	if(charging)
+		return
+	ranged_cooldown = world.time + ranged_cooldown_time
+
+	if(client)
+		if(chosen_attack == 1)
+			create_legion_skull()
+		else if(chosen_attack == 2)
+			charge_target()
+		return
+
+	if(prob(75))
+		create_legion_skull()
+	else
+		charge_target()
+
+/mob/living/simple_animal/hostile/megafauna/legion/proc/create_legion_skull()
+	var/mob/living/simple_animal/hostile/asteroid/hivelordbrood/legion/A = new(loc)
+	A.GiveTarget(target)
+	A.friends = friends
+	A.faction = faction
+
+/mob/living/simple_animal/hostile/megafauna/legion/proc/charge_target()
+	visible_message("<span class='warning'><b>[src] charges!</b></span>")
+	SpinAnimation(speed = 20, loops = 5)
+	ranged = 0
+	retreat_distance = 0
+	minimum_distance = 0
+	speed = 0
+	charging = 1
+	addtimer(CALLBACK(src, .proc/reset_charge), 50)
```

Comment:
some of these look like boos

Reply:
i didnt make the code for these i just moved them to their own procs so i’m not really sure what you’re referring to

all i can see is charging should be true

Reply:
yep